build/patches/experimental-user-scripts-support.patch

```diff
@@ -0,0 +1,10299 @@
+--- a/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadUtils.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadUtils.java
+@@ -63,6 +63,7 @@ import org.chromium.content_public.browser.BrowserStartupController;
+ import org.chromium.content_public.browser.LoadUrlParams;
+ import org.chromium.ui.base.DeviceFormFactor;
+ import org.chromium.ui.widget.Toast;
++import org.chromium.chrome.browser.user_scripts.UserScriptsUtils;
+ 
+ import java.io.File;
+ 
+@@ -382,6 +383,11 @@ public class DownloadUtils {
+         Context context = ContextUtils.getApplicationContext();
+         DownloadManagerService service = DownloadManagerService.getDownloadManagerService();
+ 
++        if (UserScriptsUtils.getInstance().openFile(filePath, mimeType, downloadGuid,
++                            isOffTheRecord, originalUrl, referrer)) {
++            return true;
++        }
++
+         // Check if Chrome should open the file itself.
+         if (service.isDownloadOpenableInBrowser(isOffTheRecord, mimeType)) {
+             // Share URIs use the content:// scheme when able, which looks bad when displayed
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java b/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
+--- a/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
+@@ -120,6 +120,8 @@ public class MainSettings extends PreferenceFragmentCompat
+             findPreference(PREF_PRIVACY).setTitle(R.string.prefs_privacy_security);
+         }
+ 
++        createBromitePreferences();
++
+         cachePreferences();
+ 
+         updatePasswordsPreference();
+@@ -274,4 +276,9 @@ public class MainSettings extends PreferenceFragmentCompat
+             }
+         };
+     }
++
++    private void createBromitePreferences() {
++        SettingsUtils.inflatePrefsAddon(this, "main_preferences_addon_",
++            R.xml.class.getFields());
++    }
+ }
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/settings/SettingsActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/settings/SettingsActivity.java
+--- a/chrome/android/java/src/org/chromium/chrome/browser/settings/SettingsActivity.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/settings/SettingsActivity.java
+@@ -274,9 +274,13 @@ public class SettingsActivity extends ChromeBaseAppCompatActivity
+             finish();
+             return true;
+         } else if (item.getItemId() == R.id.menu_id_general_help) {
+-            HelpAndFeedbackLauncherImpl.getInstance().show(this,
+-                    getString(R.string.help_context_settings), Profile.getLastUsedRegularProfile(),
+-                    null);
++            if (mainFragment instanceof SettingsUtils.ISupportHelpAndFeedback) {
```

Comment:
Perhaps we should remove this in the other Bromite patch?

Reply:
yes, it's possibile.
as I wrote: "Last note, the patch should be divided into 4.."

Reply:
Ok, let's do that then?

Reply:
yes

Reply:
fixed in https://github.com/bromite/bromite/pull/857/commits/8c164138391082350f20493a0ca5c4926887372c

Reply:
no, one at a time, only the order is not fixed